ansible_runner/runner_config.py

```diff
@@ -705,22 +705,88 @@ def _parse_cli_execenv_cmd_playbook_args():
                                 )
                             ])
 
-            # volume mount ~/.ssh/ into the exec env container
-            new_args.extend(["-v", "{}/.ssh/:/home/runner/.ssh/".format(os.environ['HOME'])])
-
-            # volume mount system-wide ssh_known_hosts the exec env container
-            if os.path.exists('/etc/ssh/ssh_known_hosts'):
-                new_args.extend(["-v", "/etc/ssh/ssh_known_hosts:/etc/ssh/ssh_known_hosts"])
-
-            # handle ssh-agent "forwarding" into the exec env container
-            if 'SSH_AUTH_SOCK' in os.environ:
-                new_args.extend(
-                    ["-v", "{}:{}".format(
-                        os.path.dirname(os.environ['SSH_AUTH_SOCK']), 
-                        os.path.dirname(os.environ['SSH_AUTH_SOCK'])
-                    )]
-                )
-                new_args.extend(["-e", "SSH_AUTH_SOCK={}".format(os.environ['SSH_AUTH_SOCK'])])
+
+            # Handle automounts
+            cli_automounts = [
+                {
+                    'ENVS': ['SSH_AUTH_SOCK'],
+                    'PATHS': [
+                        {
+                            'src': '{}/.ssh/'.format(os.environ['HOME']),
+                            'dest': '/home/runner/.ssh/'
+                        },
+                        {
+                            'src': '/etc/ssh/ssh_known_hosts',
+                            'dest': '/etc/ssh/ssh_known_hosts'
+                        }
+                    ]
+                },
+                {
+                    "ENVS": ['K8S_AUTH_KUBECONFIG'],
+                    "PATHS": [
+                        {
+                            'src': '{}/.kube/'.format(os.environ['HOME']),
+                            'dest': '/home/runner/.kube/'
+                        },
+                    ]
+                },
+                {
+                    "ENVS": [
+                        'AWS_URL', 'EC2_URL', 'AWS_ACCESS_KEY_ID', 'AWS_ACCESS_KEY',
+                        'EC2_ACCESS_KEY', 'AWS_SECRET_ACCESS_KEY', 'AWS_SECRET_KEY', 'EC2_SECRET_KEY',
+                        'AWS_SECURITY_TOKEN', 'EC2_SECURITY_TOKEN', 'AWS_REGION', 'EC2_REGION'
```

Comment:
Background question here, for the `ansible-runner playbook` command, does it copy all the env vars from the host into the container?

For environment keys that point to a path on the filesystem, couldn't you just loop over all the host environment variables, and if it points to a valid path, mount that. Unless it's something that shouldn't be allowed? I just think it would be easier to build out an exclude list, rather than building out an allow list.

Reply:
@AlanCoding I hadn't considered that. The main reason I was trying the include-list approach was to make sure we're not mounting too many things into the container and/or accidentally adding a ENV var or mount point that has unexpected consequences.